runtime/gc_vlhgc/CardListFlushTask.cpp

```diff
@@ -118,9 +121,12 @@ MM_CardListFlushTask::run(MM_EnvironmentBase *envBase)
 					while(0 != (card = rsclCardIterator.nextReferencingCard(env))) {
 						/* For Marking purposes we do not need to track references within Collection Set */
 						MM_HeapRegionDescriptorVLHGC *referencingRegion = interRegionRememberedSet->tableDescriptorForRememberedSetCard(card);
-						if (referencingRegion->containsObjects() && !referencingRegion->_markData._shouldMark) {
-							Card *cardAddress = interRegionRememberedSet->rememberedSetCardToCardAddr(env, card);
-							writeFlushToCardState(cardAddress, gmpIsActive);
+						if (interRegionRememberedSet->cardMayContainObjects(card, referencingRegion, needToCheckMarkMap, markMap) && !referencingRegion->_markData._shouldMark) {
+							bool needToFlush = true;
```

Comment:
no need for needToFlush flag, any more. here, and at a couple of other similar spots.